src/braket/default_simulator/result_types.py

```diff
@@ -294,10 +281,10 @@ def __init__(self, observable: Observable):
         super().__init__(observable)
 
     @staticmethod
-    def _calculate_from_prob_distribution(
-        probabilities: np.ndarray, eigenvalues: np.ndarray
-    ) -> float:
-        return (probabilities @ eigenvalues).real
+    def _calculate_from_application(
+        simulation: Simulation, apply_func: Callable[[np.ndarray], np.ndarray]
+    ):
+        return simulation.expectation(apply_func(simulation.state_as_tensor))
```

Comment:
I don't understand how this works after looking at `expectation`. Can you elaborate?

Reply:
Implements abstract method in `ObservableResultType`; see docstring there